lib/IDE/Refactoring.cpp

```diff
@@ -6640,17 +6769,80 @@ class AsyncConverter : private SourceEntityWalker {
       llvm_unreachable("Invalid top handler");
     }
 
+    // If ArgNames is not None, then the arguments to the completion handler
+    // were unwrapped using guard-let into variables with these names.
+    Optional<SmallVector<StringRef, 4>> ArgNames;
+
+    if (TopHandler.isAmbiguousCallToParamHandler(CE)) {
+      ArgNames.emplace();
+      // If the completion handler call was ambiguous, then all of its
+      // parameters were most likely of Optional type - otherwise, one of the
+      // parameters would always have to be `nil` or both an error and a success
+      // are passed back.
+      // Since the continuation doesn't take Optional values, we need to unwrap
+      // them.
+      unsigned I = 0;
+      for (auto Arg : Args) {
+        StringRef ArgName;
+        if (auto DRE = dyn_cast<DeclRefExpr>(Arg)) {
+          ArgName = DRE->getDecl()->getBaseIdentifier().str();
+        } else if (auto IIOE = dyn_cast<InjectIntoOptionalExpr>(Arg)) {
+          if (auto DRE = dyn_cast<DeclRefExpr>(IIOE->getSubExpr())) {
+            ArgName = DRE->getDecl()->getBaseIdentifier().str();
+          }
+        }
+        if (ArgName.empty()) {
+          if (Args.size() == 1) {
+            // We only have a single result. 'result' seems a resonable name.
+            ArgName = createUniqueName("result").str();
+          } else {
+            // We are returning a tuple. Name the result elements 'result' +
+            // index in tuple.
+            ArgName = createUniqueName("result" + std::to_string(I)).str();
```

Comment:
Should this be changed to be `I + 1` after the indexing change? IMO that's a little more natural for naming. It seems we also don't have a test case that exercises this, would it be worth adding one?

Reply:
That’s why I had `I` be `1`-based initially. But then I realized that if you have a tuple `let result: (Int, Int)`, you also access it `0`-based (e.g. `result.0`), so I thought that it’s also natural to use `0`-based indexing here.

Reply:
And I thought I added a test case for this but apparently but apparently I didn’t. Did so now.

Reply:
Hmm, yeah that's a fair point, I don't feel too strongly about it either way

Reply:
Let’s leave it `0`-based then. I think it looks rather pretty in the new test case
```swift
continuation.resume(returning: (result0, result1))
```